### PR TITLE
Docs: Add section to enable `--auth` for production in start command

### PR DESCRIPTION
**Pull request by Ekwuno**

### CODE REVIEW

File: app/templates/docs/cli/start.hbs [hunk 1]
```diff
@@ -120,6 +127,12 @@
 	<Code @name="docs-releases-1-0-0-beta-4-strict.shell">
 		user@localhost % surreal start --strict --log debug memory
 	</Code>
+	<h4>Authentication</h4>
+	<p>SurrealDB supports the ability to startup with authentication enabled. When running with authentication enabled, all requests to the database will require a valid root username and password.</p>
+	<blockquote yellow text="Note"> It is adviced to enable <code>--auth</code> for production mode.</blockquote>
+	<Code @name="docs-releases-1-0-0-beta-4-auth.shell">
```
Comment (kearfy) [inline app/templates/docs/cli/start.hbs:133]: the name for this codeblock seems off

File: app/templates/docs/cli/start.hbs [hunk 2]
```diff
@@ -65,6 +65,13 @@
 					</td>
 					<td>Sets the master password for the database</td>
 				</tr>
+				<tr>
+					<td>
+						<code>--auth</code>
+						<l faint r>Optional</l>
+					</td>
+					<td>enables authentication for the database</td>
```
Comment (kearfy) [inline app/templates/docs/cli/start.hbs:73]:
```suggestion
					<td>Enables authentication for the database</td>
```

File: app/templates/docs/cli/start.hbs [hunk 3]
```diff
@@ -120,6 +127,12 @@
 	<Code @name="docs-releases-1-0-0-beta-4-strict.shell">
 		user@localhost % surreal start --strict --log debug memory
 	</Code>
+	<h4>Authentication</h4>
+	<p>SurrealDB supports the ability to startup with authentication enabled. When running with authentication enabled, all requests to the database will require a valid root username and password.</p>
```
Comment (kearfy) [inline app/templates/docs/cli/start.hbs:131]:
```suggestion
	<p>Authentication mode is not enabled by default in SurrealDB. When not enabled, any guest user is considered to have the same permissions as a root user with the <code>OWNER</code> role.</p>
```

File: app/templates/docs/cli/start.hbs [hunk 4]
```diff
@@ -120,6 +127,12 @@
...
+	<blockquote yellow text="Note"> It is adviced to enable <code>--auth</code> for production mode.</blockquote>
```
Comment (kearfy) [inline app/templates/docs/cli/start.hbs:132]:
```suggestion
	<blockquote yellow text="Note"> It is strongly recommended to enable <code>--auth</code> mode when deploying SurrealDB in production. Not having it enabled can result in unauthorised access to your database.</blockquote>
```

File: app/templates/docs/cli/start.hbs (hunk 1 above)
Comment (Ekwuno) [reply app/templates/docs/cli/start.hbs:133]: Same as the others but really won't need the name files in a bit 😉

Comment (kearfy) [reply app/templates/docs/cli/start.hbs:133]: haha true, I guess that as long as it doesn't conflict it will be fine

**Review by kearfy — APPROVED** [review]
LGTM!

**Pull request merged**